superbench/benchmarks/micro_benchmarks/directx_render_performance/BufferHelper.h

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <random>
+
+#include "../directx_third_party/DXSampleHelper.h"
+#include "../directx_third_party/d3dx12.h"
+
+// Helper class for creating and uploading resources to the GPU.
+template <typename T> class UploadBuffer {
+  public:
+    UploadBuffer(ID3D12Device *device, UINT elementCount, bool isConstantBuffer)
+        : m_isConstantBuffer(isConstantBuffer) {
+        m_elementByteSize = sizeof(T);
+
+        if (isConstantBuffer)
+            m_elementByteSize = CalcConstantBufferByteSize(sizeof(T));
+
+        ThrowIfFailed(
```

Comment:
>ThrowIfFailed [](http://example.com/codeflow?start=8&length=13)

Looks like it will throw exception in the constructor, so do we check it when calling it?

Reply:
added a try...catch... in the main loop